cmd/kube-apiserver/app/server.go

```diff
@@ -367,9 +370,12 @@ func BuildMasterConfig(s *options.ServerRunOptions) (*master.Config, informers.S
 		return nil, nil, err
 	}
 
-	clientCA, err := readCAorNil(s.Authentication.ClientCert.ClientCA)
-	if err != nil {
-		return nil, nil, err
+	clientCA := s.Authentication.ClientCert.ClientCAData
```

Comment:
Would seem cleaner to switch readCAorNil to readCAorDefault. Use nil as the default for existing cases and s.Authentication.ClientCert.ClientCAData as the default for this case.